mono/metadata/class-init.h

```diff
@@ -0,0 +1,80 @@
+/**
+ * \file
+ * Copyright 2018 Microsoft
+ * Licensed under the MIT license. See LICENSE file in the project root for full license information.
+ */
+#ifndef __MONO_METADATA_CLASS_INIT_H__
+#define __MONO_METADATA_CLASS_INIT_H__
+
+#include <glib.h>
+#include <mono/metadata/metadata.h>
+
+MONO_BEGIN_DECLS
+
+void
+mono_classes_init (void);
+
+void
+mono_classes_cleanup (void);
+
+MonoClass *
+mono_class_create_from_typedef (MonoImage *image, guint32 type_token, MonoError *error);
```

Comment:
Could we rename all the functions in this header to something like `mono_class_{init,create}_*`. For example they could be in order:
 - mono_class_create_from_typedef
 - mono_class_create_from_generic_class
 - mono_class_create_bounded_array
 - mono_class_create_array
 - mono_class_create_from_generic_parameter_internal
 - mono_class_create_ptr
 - mono_class_create_fnptr
 - mono_class_init_vtable_general
 - mono_class_init_sizes
 - mono_class_init_basic_field_info
 - ...

Reply:
And for the functions that are MONO_API, we can just keep their old version in `class.c` and have them call the new version in `class-init.c`.

Reply:
Sure. Normalizing the names is a great idea.  I would prefer to do keep this PR focused on just moving functions to a separate file as much as possible, and do a followup for renaming.

Reply:
Oh, but a few of these are new in this PR. I'll update those to use the `mono_class_create_foo`/`mono_class_init_foo` naming.

Reply:
Ok, renamed all the creation functions to be `mono_class_create_XXX`.  The init/setup ones I think I still want to do in a separate PR, I think.